Library/ViewModels/WatchProjectViewModel.swift

```diff
@@ -102,10 +102,8 @@ public final class WatchProjectViewModel: WatchProjectViewModelType,
       .switchMap { project, shouldWatch in
         watchProjectProducer(with: project, shouldWatch: shouldWatch)
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-          .map { $0.watchProject.project.isWatched }
           .map { _ in (project, project.personalization.isStarred ?? false, success: true) }
           .flatMapError { _ in .init(value: (project, !shouldWatch, success: false)) }
-          .take(until: saveButtonTapped.ignoreValues())
```

Comment:
What was the motivation for removing this line? I'm not super clear on why it was there at all, maybe @justinswart can shed some light on this.

Reply:
When I saw this code I couldn't understand either why we had those lines, and removing them seems that didn't change anything. @justinswart any idea why we had those?

Reply:
Hmm yeah this is some time ago, line `105` seems strange, I'm not sure why I was mapping that value and then ignoring it 🤷‍♂so you can probably remove that. I imagine the `take(until:)` was there because I wanted the producer to immediately complete if the button was tapped again. You may look at the code and think that we don't need that because the `switchMap` would take care of that but take note of the debouncing that's happening. It will take 500ms before the switchMap cancels the in-flight request so I think we do still want that `take(until:)` so that it completes/cancels immediately.

Nevertheless, making any changes here is going to require some regression testing!